pkg/imgpkg/registry/keychain.go

```diff
@@ -49,7 +50,12 @@ func (k *envKeychain) Resolve(target regauthn.Resource) (regauthn.Authenticator,
 	}
 
 	for _, info := range infos {
-		if info.Hostname == target.RegistryStr() {
+		registryURLMatches, err := credentialprovider.URLsMatchStr(info.Hostname, target.RegistryStr())
```

Comment:
might be good to document for users (lightly) how URLsMatchStr behaves.

Reply:
agreed. I had a TODO to document the new behavior of `IMGPKG_REGISTRY_HOSTNAME`

Reply:
nah, just regular docs on carvel.dev